File: src/main/java/com/back/pinco/domain/user/dto/UserDto.java
package com.back.pinco.domain.user.dto;

import com.back.pinco.domain.user.entity.User;

import java.time.LocalDateTime;

public record UserDto(
        Long id,
        String email,
        String userName,
        String password,
        LocalDateTime createdAt,
        LocalDateTime modifiedAt
) {
    public UserDto(User user) {
        this(
                user.getId(),
                user.getEmail(),
                user.getUserName(),
                user.getPassword(),
                user.getCreatedAt(),
                user.getModifiedAt()
        );
    }
}

File: src/main/java/com/back/pinco/domain/user/dto/UserDto.kt
package com.back.pinco.domain.user.dto

import com.back.pinco.domain.user.entity.User
import java.time.LocalDateTime

data class UserDto(
    val id: Long,
    val email: String,
    val userName: String,
    val password: String,
    val createdAt: LocalDateTime,
    val modifiedAt: LocalDateTime
) {
    constructor(user: User) : this(
        user.id!!,
        user.email,
        user.userName,
        user.password,
        user.createdAt!!,
        user.modifiedAt!!
    )
}

